docling/pipeline/vlm_pipeline.py

```diff
@@ -139,25 +139,23 @@ def _assemble_document(self, conv_res: ConversionResult) -> ConversionResult:
                 conv_res.document.load_from_doctags(doctags_doc)
 
                 # If forced backend text, replace model predicted text with backend one
-                if page.size:
```

Comment:
page is currently the last page from the above loop, which is incorrect and needs to be fetched for each element using element.prov[0].page_no - 1.